### PR TITLE
[FVM] removing legacy controller from storage

### PR DESCRIPTION
**Pull request by ramtinms**
This PR
  - continues the work at #2585 
  - removes the controller from storage

remaining work
- [x] update EN RPC proto at flow repo and bump the version here
- [x] update flow emulator and update the dkg code to use the new one

### CODE REVIEW

**Review by fxamacker — COMMENTED** [review]
Looks great!  Thanks for removing controller reference in all these files!

Left some comments related to register key `String()` but most comments are about tests.

File: cmd/util/ledger/migrations/account_status_migration_test.go [hunk 1]
```diff
@@ -48,8 +39,9 @@ func TestAccountStatusMigration(t *testing.T) {
 	require.True(t, newPayloads[2].Equals(&payloads[2]))
 
 	expectedPayload := &ledger.Payload{
-		Key:   createAccountPayloadKey(address1, state.KeyAccountStatus),
+		Key:   createPayloadKeyWithLegacyController(address1, state.KeyAccountStatus, true),
 		Value: state.NewAccountStatus().ToBytes(),
 	}
+	fmt.Println(newPayloads[3])
```
[the remark this answers is not in the document]
Comment (ramtinms) [reply cmd/util/ledger/migrations/account_status_migration_test.go]: nice catch, removed at #3f13a66

File: cmd/util/ledger/migrations/legacy_controller_migration_test.go [hunk 2]
```diff
@@ -32,11 +41,11 @@ func TestLegacyControllerMigration(t *testing.T) {
...
-		{Key: createPayloadKeyWithLegacyController(address2, state.KeyPublicKey(1)), Value: utils.Uint64ToBinary(4)},
-		{Key: createPayloadKeyWithLegacyController(address2, state.KeyPublicKeyCount), Value: utils.Uint64ToBinary(4)},
+		{Key: createPayloadKeyWithLegacyController(address1, state.KeyStorageUsed, false), Value: utils.Uint64ToBinary(1)},
+		{Key: createPayloadKeyWithLegacyController(address1, state.ContractKey("CoreContract"), true), Value: utils.Uint64ToBinary(2)},
+		{Key: createPayloadKeyWithLegacyController(address1, state.KeyContractNames, true), Value: utils.Uint64ToBinary(3)},
+		{Key: createPayloadKeyWithLegacyController(address2, state.KeyPublicKey(1), true), Value: utils.Uint64ToBinary(4)},
+		{Key: createPayloadKeyWithLegacyController(address2, state.KeyPublicKeyCount, true), Value: utils.Uint64ToBinary(4)},
 	}
 
 	newPayloads, err := mig.Migrate(payloads)
```
[the remark this answers is not in the document]
Comment (ramtinms) [reply cmd/util/ledger/migrations/legacy_controller_migration_test.go:68]: good suggestion, updated the tests as recommended at #3f13a66

File: engine/execution/state/delta/delta_test.go [hunk 3]
```diff
@@ -121,11 +121,11 @@ func TestDelta_RegisterUpdatesAreSorted(t *testing.T) {
...
-	data[0].Key = flow.NewRegisterID("a", "a", "1")
-	data[1].Key = flow.NewRegisterID("a", "b", "1")
-	data[2].Key = flow.NewRegisterID("b", "a", "1")
-	data[3].Key = flow.NewRegisterID("b", "b", "1")
-	data[4].Key = flow.NewRegisterID("b", "b", "2")
+	data[0].Key = flow.NewRegisterID("a", "1")
+	data[1].Key = flow.NewRegisterID("b", "1")
+	data[2].Key = flow.NewRegisterID("c", "1")
+	data[3].Key = flow.NewRegisterID("d", "1")
+	data[4].Key = flow.NewRegisterID("e", "2")
```
[the remark this answers is not in the document]
Comment (ramtinms) [reply engine/execution/state/delta/delta_test.go]: good point, updated at 3f13a66

File: model/flow/ledger.go [hunk 4]
```diff
@@ -10,30 +10,24 @@ import (
...
 
-// this function returns a string format of a RegisterID in the form '%x/%x/%x'
+// this function returns a string format of a RegisterID in the form '%x/%x'
 // it has been optimized to avoid the memory allocations inside Sprintf
 func (r *RegisterID) String() string {
 	ownerLen := len(r.Owner)
-	controllerLen := len(r.Controller)
 
-	requiredLen := ((ownerLen + controllerLen + len(r.Key)) * 2) + 2
+	requiredLen := ((ownerLen + len(r.Key)) * 2) + 2
```
[the remark this answers is not in the document]
Comment (ramtinms) [reply model/flow/ledger.go]: updated at 3f13a66

File: model/flow/ledger_test.go [hunk 5]
```diff
@@ -16,27 +16,21 @@ var length int
...
-		Key:        "123412341234",
+		Owner: "theowner",
+		Key:   "123412341234",
 	}
 
 	ownerLen := len(r.Owner)
-	controllerLen := len(r.Controller)
 
-	requiredLen := ((ownerLen + controllerLen + len(r.Key)) * 2) + 2
+	requiredLen := ((ownerLen + len(r.Key)) * 2) + 2
```
[the remark this answers is not in the document]
Comment (ramtinms) [reply model/flow/ledger_test.go]: updated at 3f13a66

**Review by fxamacker — APPROVED** [review]

File: cmd/util/ledger/migrations/legacy_controller_migration_test.go [hunk 6]
```diff
@@ -23,6 +32,15 @@ func createPayloadKeyWithLegacyController(a flow.Address, key string) ledger.Key
 	}
 }
 
+func createMigratedPayloadKey(a flow.Address, key string) ledger.Key {
+	return ledger.Key{
+		KeyParts: []ledger.KeyPart{
+			ledger.NewKeyPart(0, a.Bytes()),
+			ledger.NewKeyPart(2, []byte(key)),
```
Comment (janezpodhostnik) [inline cmd/util/ledger/migrations/legacy_controller_migration_test.go:39]: Can we use `KeyPartOwner | KeyPartController | KeyPartKey` here?

**Review by janezpodhostnik — APPROVED** [review]
Looks good to me.

Great to have this removed finally :).

File: cmd/util/ledger/migrations/legacy_controller_migration_test.go (hunk 6 above)
Comment (ramtinms) [reply cmd/util/ledger/migrations/legacy_controller_migration_test.go]: fixed at 6706c4d

**Pull request merged**